src/lib/services/BidService.js

```diff
@@ -83,6 +83,19 @@ export default class BidService {
     const result = { bidGroup: null, error: null }
 
     if (utils.isEmptyObject(parcelStates.error)) {
+      for (let index in bidGroup.bids) {
+        const bid = bidGroup.bids[index]
+        const parcelId = this.ParcelState.hashId(bid.x, bid.y)
+        const parcelState = parcelMap.find(parcel => parcel.id === parcelId)
+        if (parcelState.address && parcelState.address !== bidGroup.address) {
+          const outbidAddress = await this.AddressState.findByAddress(parcelState.address)
+          outbidAddress.balance = getBn(outbidAddress.balance)
+            .plus(getBn(parcelState.amount))
+            .toString()
+          await this.AddressState.update(outbidAddress, { id: outbidAddress.id })
```

Comment:
Two things:

- Why don't we add a test case for this situation?
- It might be a good idea to extract it to a method with an appropriate name (also easier to test)
- I should take a closer look but at first sight it looks like it's looping every bid and adding the balance to the losers. But what happens when the same parcel is outbid more than once by different people? Wouldn't this be adding the `parcelState` amount multiple times?

Reply:
* +1 test
* +1 on it
* this is only for one person outbidding others